Include/internal/pycore_runtime_init.h

```diff
@@ -93,26 +93,34 @@ extern "C" {
         _PyBytes_SIMPLE_INIT(CH, 1) \
     }
 
-#define _PyASCIIObject_INIT(LITERAL) \
+#define _PyASCIIObjectBase_INIT(LITERAL, ASCII) \
     { \
-        ._ascii = { \
-            .ob_base = _PyObject_IMMORTAL_INIT(&PyUnicode_Type), \
-            .length = sizeof(LITERAL) - 1, \
-            .hash = -1, \
-            .state = { \
-                .kind = 1, \
-                .compact = 1, \
-                .ascii = 1, \
-                .ready = 1, \
-            }, \
+        .ob_base = _PyObject_IMMORTAL_INIT(&PyUnicode_Type), \
+        .length = sizeof(LITERAL) - 1, \
+        .hash = -1, \
+        .state = { \
+            .kind = 1, \
+            .compact = 1, \
+            .ascii = ASCII, \
+            .ready = 1, \
         }, \
-        ._data = LITERAL, \
+    }
+#define _PyASCIIObject_INIT(LITERAL) \
+    { \
+        ._ascii = _PyASCIIObjectBase_INIT(LITERAL, 1), \
+        ._data = LITERAL \
     }
 #define INIT_STR(NAME, LITERAL) \
     ._ ## NAME = _PyASCIIObject_INIT(LITERAL)
 #define INIT_ID(NAME) \
     ._ ## NAME = _PyASCIIObject_INIT(#NAME)
-
+#define _PyCompactUnicodeObject_INIT(LITERAL) \
```

Comment:
The name here might imply the wrong thing (since there are compact strings other than latin1) , so consider something more specific to the use case:

```suggestion
#define _PyUnicode_LATIN1_BASE_INIT(LITERAL) \
```

Reply:
Fixed